test/FunctionalTests/Microsoft.OData.Core.Tests/ValidationHelper.cs

```diff
@@ -0,0 +1,117 @@
+                    }
+                }
+                else if (typeof(IComparable).IsAssignableFrom(prop.PropertyType) || prop.PropertyType.IsPrimitive() || prop.PropertyType.IsValueType())
+                {
+                    if ((!val1.Equals(val2)))
+                    {
+                        diff.Add(string.Format("Value of {0} does not match.", prop.Name));
+                    }
+                }
+                else if (prop.PropertyType.IsClass())
+                {
+                    EvaluateDifferences(val1, val2, diff);
+                }
+                else if (typeof(Enumerable).IsAssignableFrom(prop.PropertyType))
+                {
+                    var collection1 = ((IEnumerable)val1).Cast<object>();
+                    var collection2 = ((IEnumerable)val2).Cast<object>();
+
+                    if (collection1.Count() != collection2.Count())
+                    {
+                        diff.Add(string.Format("Collection count of {0} does not match.", prop.Name));
+                    }
+                    else
+                    {
+                        for (int i = 0; i < collection1.Count(); i++)
+                        {
+                            var element1 = collection1.ElementAt(i);
+                            var element2 = collection2.ElementAt(i);
+                            var collpropType = element1.GetType();
+
+                            if (element1 == null && element2 == null)
+                            {
+                                continue;
+                            }
+
+                            if (element1 == null || element2 == null)
+                            {
+                                diff.Add(string.Format("Value of items in Collection {0} does not match.", prop.Name));
+                            }
+                            else if (typeof(Uri) == collpropType)
+                            {
+                                if (Uri.Compare((Uri)element1, (Uri)element2, UriComponents.AbsoluteUri, UriFormat.Unescaped, StringComparison.CurrentCulture) != 0)
+                                {
+                                    diff.Add(string.Format("Value of Uri collection {0} does not match.", prop.Name));
+                                }
+                            }
+                            else if (typeof(IComparable).IsAssignableFrom(collpropType) || collpropType.IsPrimitive() || collpropType.IsValueType())
+                            {
+                                if ((!element1.Equals(element2)))
+                                {
+                                    diff.Add(string.Format("Values of Collection {0} does not match.", prop.Name));
+                                }
+                            }
+                            else
+                            {
+                                EvaluateDifferences(element1, element2, diff);
+                            }

```

Comment:
It seems you can replace all by calling EvaluateDifferences again?

Reply:
The text for this contains collection, eg: "Values of Collection {0} does not match.". So to keep that text and any further modificatioins needed for collection did like this. Pls let me know if thats fine

Reply:
Resolved